packages/server/lib/controllers/v1/environment/postEnvironment.ts

```diff
@@ -0,0 +1,51 @@
+import { z } from 'zod';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+import type { PostEnvironment } from '@nangohq/types';
+import { environmentService, externalWebhookService } from '@nangohq/shared';
+import { envSchema } from '../../../helpers/validation.js';
+
+const validationBody = z
+    .object({
+        name: envSchema
+    })
+    .strict();
+
+export const postEnvironment = asyncWrapper<PostEnvironment>(async (req, res) => {
+    const emptyQuery = requireEmptyQuery(req);
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const valBody = validationBody.safeParse(req.body);
+    if (!valBody.success) {
+        res.status(400).send({ error: { code: 'invalid_body', errors: zodErrorToHTTP(valBody.error) } });
+        return;
+    }
+
+    const body: PostEnvironment['Body'] = valBody.data;
+
+    const accountId = res.locals.user.account_id;
+
+    const exists = await environmentService.getAccountAndEnvironment({ accountId, envName: body.name });
+    if (exists) {
+        res.status(400).send({ error: { code: 'invalid_body', message: 'Environment already exists' } });
+        return;
+    }
+
+    const created = await environmentService.createEnvironment(accountId, body.name);
+    if (!created) {
+        res.status(500).send({ error: { code: 'server_error', message: 'Failed to create environment' } });
+        return;
+    }
+
+    await externalWebhookService.update(created.id, {
+        alwaysSendWebhook: true,
+        sendAuthWebhook: true,
+        sendRefreshFailedWebhook: true,
+        sendSyncFailedWebhook: true
+    });
```

Comment:
can this fail? what should we do if it does?

Reply:
I guess it can fail, like the previous message it will throw and return a 500, which is not very different than catching it and returning a 500 imo

Reply:
as I said I prefer explicit error handling in general but here at least we are at the endpoint level (aka most outer level) so I think it is ok to be more implicit